src/analysis/analytic/uniqueness.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import analysis.analytic.linear
+import analysis.analytic.composition
+import analysis.normed_space.completion
+
+/-!
+# Uniqueness principle for analytic functions
+
+We show that two analytic functions which coincide around a point coincide on whole connected sets,
+in `analytic_on.eq_on_of_preconnected_of_eventually_eq`.
+-/
+
+variables {𝕜 : Type*} [nontrivially_normed_field 𝕜]
+{E : Type*} [normed_add_comm_group E] [normed_space 𝕜 E]
+{F : Type*} [normed_add_comm_group F] [normed_space 𝕜 F]
+
+open set
+open_locale topological_space ennreal
+
+namespace analytic_on
+
+/-- If an analytic function vanishes around a point, then it is uniformly zero along
+a connected set. Superseded by `eq_on_zero_of_preconnected_of_locally_zero` which does not assume
+completeness of the target space. -/
+theorem eq_on_zero_of_preconnected_of_eventually_eq_zero_aux [complete_space F]
+  {f : E → F} {U : set E} (hf : analytic_on 𝕜 f U) (hU : is_preconnected U) {w : E}
+  (hw : w ∈ U) (hfw : f =ᶠ[𝓝 w] 0) : eq_on f 0 U :=
+begin
+  /- Let `u` be the set of points around which `f` vanishes. It is clearly open. We have to show
+  that its limit points in `U` still belong to it, from which the inclusion `U ⊆ u` will follow
+  by connectedness.
+  Take a limit point `x`, then a ball `B (x, r)` on which it has a power series expansion, and
+  then `y ∈ B (x, r/2) ∩ u`. Then `f` has a power series expansion on `B (y, r/2)` as it is
+  contained in `B (x, r)`. All the coefficients in this series expansion vanish, as `f` is zero on a
+  neighborhood of `y`. Therefore, `f` is zero on `B (y, r/2)`. As this ball contains `x`, it follows
+  that `f` vanishes on a neighborhood of `x`, proving the claim. -/
+  let u := {x | f =ᶠ[𝓝 x] 0},
+  have main : closure u ∩ U ⊆ u,
+  { rintros x ⟨xu, xU⟩,
+    rcases hf x xU with ⟨p, r, hp⟩,
+    obtain ⟨y, yu, hxy⟩ : ∃ y ∈ u, edist x y < r / 2,
+      from emetric.mem_closure_iff.1 xu (r / 2) (ennreal.half_pos hp.r_pos.ne'),
+    let q := p.change_origin (y - x),
+    have has_series : has_fpower_series_on_ball f q y (r / 2),
+    { have A : (∥y - x∥₊ : ℝ≥0∞) < r / 2, by rwa [edist_comm, edist_eq_coe_nnnorm_sub] at hxy,
+      have := hp.change_origin (A.trans_le ennreal.half_le_self),
+      simp only [add_sub_cancel'_right] at this,
+      apply this.mono (ennreal.half_pos hp.r_pos.ne'),
+      apply ennreal.le_sub_of_add_le_left ennreal.coe_ne_top,
+      apply (add_le_add (A.le) (le_refl (r / 2))).trans (le_of_eq _),
+      exact ennreal.add_halves _ },
```

Comment:
Yeah this part is painful, perhaps it would be useful to factor it out as an `ennreal` (or `with_top`) lemma.

Reply:
As I am already used to manipulating ennreals, I did this part on automatic pilot, so I didn't feel it as painful. What kind of lemmas would you have in mind?

Reply:
My autopilot does not work on ennreal yet, so it was painful for me :-( but just a lemma showing that `a < b/2` implies `b/2 < b - a` (or a proper generalization of it) could probably be used at many places. It would go together well with `ennreal.add_lt_add_left` and the like.